### DISCUSSION TITLE
Ubuntu開發機環境：虛擬機網路無法連線

### DISCUSSION THREAD

**Discussion by asdshawn** [bol-edu/HLS-SOC-Discussions#58] [Q&A] [answered]
我在主機網路管理員中確認有VirtualBox Host-Only Ethernet Adapter並有啟用DHCP，
也已經在Ununtu的介面卡2有選取，
但是開啟虛擬機時仍無法連線，以至於無法獲取虛擬機IP位置，
想請問各位先進前輩們，對此問題有沒有解決辦法呢？
謝謝！

**Comment by boledulab** [comment]
請提供每個設定步驟的詳細截圖及資訊才能提供協助
如果是修課同學可以請TA協助

**Reply by asdshawn** [reply]
您好，感謝回復，以下是我各步驟的截圖畫面，希望提供的資訊夠完整：
1. 建立VirtualBox Host-Only Ethernet Adapter後並啟用DHCP。
![1](https://github.com/bol-edu/HLS-SOC-Discussions/assets/31488586/2c86796e-6365-4b92-8c31-9d79eba50b9e)

2. 在虛擬機網路設定中啟用"僅限主機介面卡"，並選擇VirtualBox Host-Only Ethernet Adapter。
![2](https://github.com/bol-edu/HLS-SOC-Discussions/assets/31488586/8dec3879-0d6c-48db-9bf5-e3df014d9ede)

3. 啟動虛擬機並登入後，檢查網路狀態並未成功連線。
![3](https://github.com/bol-edu/HLS-SOC-Discussions/assets/31488586/4f633be9-8be7-4c02-9655-b27a01c9ab97)

4. 終端機鍵入ifconfig也因此無法獲取IP訊息。
![4](https://github.com/bol-edu/HLS-SOC-Discussions/assets/31488586/b81dd2d8-7e1f-406a-9b20-63471386e8ef)

此外，我也會再去尋求TA的協助，謝謝您的建議。

**Reply by kevinjantw** [reply]
請再確認這個步驟是否如圖
可以是192.168.xxx.1/24
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/11850122/c56d2fa3-3c52-41a3-bdfd-9b184aa25f11)

是的話而且沒拿到ip 請再執行這兩筆指令
ifconfig enp0s8 down
ifconfig enp0s8 up
若還沒解決請完全刪除VM再重新下載操作
或是請TA協助

**Reply by asdshawn** [reply]
您好，有確認步驟和圖中相符了，如我第一步顯示的那樣。
接著，這是我執行兩筆指令後的結果：
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/31488586/7aa98442-f3f8-48a5-b650-7bfebaeba55e)

目前已嘗試過重新安裝並重啟電腦，仍然無法連線，
我會再請TA協助，
謝謝您！

**Reply by emma16163** [reply]
你好，我也有遇到相同的問題，後來發現是下載的VirtualBox和Extension Pack的版本是不相同的，版本修正之後虛擬機的網路就正常了，給你參考看看。
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/75108089/e0dc2913-885f-4c3b-8b7a-2bca429d46d0)

**Reply by asdshawn** [reply]
您好，感謝您提供解法，方才立刻去檢查VirtualBox和Extension Pack的版本，
但很遺憾發現，兩者的版本是相同的沒有錯(檔案來源均為由課堂簡報提供的載點)。
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/31488586/14032b44-fd61-4091-add7-33d5cb909966)

**Comment by kevinjantw** [comment]
請執行
sudo ifconfig enp0s8 down
sudo ifconfig enp0s8 up

也再使用系統管理員(cmd) 確認主機端的DHCP是否正常運作如下
<img width="468" alt="螢幕擷取畫面 2023-09-20 180111" src="https://github.com/bol-edu/HLS-SOC-Discussions/assets/11850122/4086db14-94de-4f8a-8d61-ea0c7e1db307">

**Reply by asdshawn** [reply]
您好，執行過兩筆指令(開關網路)後，一樣無法成功連線，
主機端的資訊如下：
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/31488586/16a7502e-f48a-4074-befe-e3fda21c3f6d)
謝謝您。

**Comment by asdshawn** [comment, marked as answer]
此問題在經過我重灌主機的作業系統，並在進行同樣的安裝操作後，得到解決。
